### PR TITLE
Add account and region configs to providers

### PR DESCRIPTION
**Pull request by ethanwe**
The goal of the PR is to allow templates that are using providers to pass through extra configuration. In this case, I wanted to allow app-params to add parameter store entries to different accounts and regions. An example of a template can be found in the diff under template.simple.yml.

I took advantage of the behavior inside of the ComponentDeploymentDecider_DeploymentHandler where the Account property is set on the provider with lookupProviderAccountFromAccountComponents. The invocation of the provider lambda is in the JobRunActor, where I added a componentProvider property containing all the necessary information.

Please take note of the change in the CodeBuild.ts file, where I set the priviledgedModeOverride to true. This was required to bundle the lambda function when deploying the app-param provider for the first time.

### CODE REVIEW

File: packages/api/src/services/CodeBuild.ts [hunk 1]
```diff
@@ -23,6 +23,7 @@ export module CodeBuildClient {
                 projectName: process.env.CODE_BUILD_PROJECT as string,
                 buildspecOverride: generateBuildSpec(params.buildDir),
                 sourceLocationOverride: `${S3_BUCKET}/${artifactGuid}.zip`,
+                privilegedModeOverride: true,
```
Comment (traviswalker26) [inline packages/api/src/services/CodeBuild.ts:26]: Why does CodeBuild need priveleged mode? Just curious.

File: packages/providers/base/assets/Provider.ts [hunk 2]
```diff
@@ -1,31 +1,70 @@
...
+    public account: Account;
+
+	public awsCredentials: Credentials;
+	public region: string;
+
     sns = new SNS();
     ssm = new SSM();
 
     constructor(event: any) {
+        console.log(event);
```
Comment (traviswalker26) [inline packages/providers/base/assets/Provider.ts:29]: We can't necessary log the event here because some providers could end up logging sensitive data.

**Review by traviswalker26 — COMMENTED** [review]
This looks really good. I did notice that in packages/providers/base/assets/Provider.ts the formatting might have gotten mixed in how many spaces were being used. Might be worth taking a closer look.

File: packages/api/src/services/CodeBuild.ts (hunk 1 above)
Comment (traviswalker26) [reply packages/api/src/services/CodeBuild.ts:26]: I missed your reason that you gave. I do know if that's true or if something changed recently. That was not a requirement previously.

**Comment by ethanwe** [comment]
> This looks really good. I did notice that in packages/providers/base/assets/Provider.ts the formatting might have gotten mixed in how many spaces were being used. Might be worth taking a closer look.

Yep! I was using tabs instead of spaces on accident. Just pushed a commit to fix that.

File: packages/providers/base/assets/Provider.ts (hunk 2 above)
Comment (ethanwe) [reply packages/providers/base/assets/Provider.ts]: I removed this log line.

File: packages/api/src/services/CodeBuild.ts (hunk 1 above)
Comment (ethanwe) [reply packages/api/src/services/CodeBuild.ts:26]: Do you think this is okay to leave here? From my understanding, it's now required if you want to bundle lambdas for cdk using docker in CodeBuild.

**Comment by ethanwe** [comment]
@traviswalker26 This is ready to be looked at again and possibly merged. Do you think the change of the naming of the app-param from {env}-{component} to /{env}/{component} will hurt anything that's already deployed?

**Review by traviswalker26 — APPROVED** [review]

**Comment by traviswalker26** [comment]
> @traviswalker26 This is ready to be looked at again and possibly merged. Do you think the change of the naming of the app-param from {env}-{component} to /{env}/{component} will hurt anything that's already deployed?

Looks good. The change in the parameter name will be fine since the name is referenced from the dynamo table so this will be backwards compatible.

File: packages/api/src/services/CodeBuild.ts (hunk 1 above)
Comment (traviswalker26) [reply packages/api/src/services/CodeBuild.ts:26]: I think this is fine then.

**Pull request merged**